src/main/java/com/squareup/kotlinpoet/LineWrapper.kt

```diff
@@ -26,71 +29,139 @@ internal class LineWrapper(
 ) {
   private var closed = false
 
-  /** Characters written since the last wrapping space that haven't yet been flushed.  */
-  private val buffer = StringBuilder()
-
-  /** The number of characters since the most recent newline. Includes both out and the buffer.  */
-  private var column = 0
+  /**
+   * Segments of the current line to be joined by spaces or wraps. Never empty, but contains a lone
+   * empty string if no data has been emitted since the last newline.
+   */
+  private val segments = mutableListOf("")
```

Comment:
Yeah, it worked out okay. I think it might get a little more complicated in future if we add a symbol for a non-space wrapping point, like JavaPoet’s `$Z`.
https://square.github.io/javapoet/1.x/javapoet/com/squareup/javapoet/CodeBlock.html